### DISCUSSION TITLE
Why has ReactQueryDevtoolsPanel been removed from v5 devtools?

### DISCUSSION THREAD

**Discussion by bazo** [TanStack/query#6564] [General]
with v4 i have integrated ReactQueryDevtoolsPanel into a custom port of symfony debug bar
![image](https://github.com/TanStack/query/assets/181588/bbf4148b-cfb5-4bde-8cb0-5dc0dd7acad9)

so the tools open in a nice popup

but in v5 this component no longer exists.
also the docs don't mention embedded mode anymore

any specific reason why this was removed?

**Comment by TkDodo** [comment]
we re-wrote the devtools from scratch and I think we just didn't migrate this feature. @ardeora do you think we could bring back embedded mode ?

**Reply by kachkaev** [reply]
@ardeora WDYT?

**Comment by k-ruben** [comment]
A solution is greatly appreciated. Current workaround:

```
const ReactQueryDevtoolsPanel = React.useRef(
        React.lazy(() =>
            import('@tanstack/react-query-devtools/build/modern/production.js').then((d) => ({
                default: d.ReactQueryDevtools,
            })),
        ),
    ).current;
```

and wrap the Component in a Suspense Boundary

```
<React.Suspense fallback={null}>
    {showReactQueryDevtoolPanel && <ReactQueryDevtoolsPanel />}
</React.Suspense>
```

**Comment by itshallrun** [comment]
Looks like this is going under. Any news?
@TkDodo

**Comment by ardeora** [comment]
I think the refactor to add the picture-in-picture mode to the devtools has unlocked the ability to add a Devtools Panel. I'll add this on my backlog

**Reply by ardeora** [reply]
If anyone wants to contribute to add the DevtoolsPanel I would be more than happy to help/assist them! I have a couple more things to fix before I can get to this atm 😄

**Reply by toofff** [reply]
I'm very interested and I propose a pull request today or this weekend to make available the ReactQueryDevTools Panel.

**Comment by toofff** [comment]
Hi :wave: 

I have a first [pull request](https://github.com/TanStack/query/pull/7788) in working draft.
I still have:
- the isOpen state to be reinstated and accessible from outside the debug bar (as before).
- add documentation
- add example

I'm planning to finish it in the next few days, so don't hesitate to tell me if I'm going in the right direction?
I could apply the same changes to the other technologies in a future pull request.

**Reply by toofff** [reply]
My pull request is finally ready :tada:

**Reply by toofff** [reply]
My pull request is merged, you will find the ReactQueryDevtoolsPanel component in release [v5.54.0](https://github.com/TanStack/query/releases/tag/v5.54.0)
